packages/shopify-app-session-storage/src/abstract-migration-engine.ts

```diff
@@ -15,15 +14,22 @@ export abstract class AbstractMigrationEngine<
   protected options: SessionStorageMigratorOptions;
   protected connection: ConnectionType;
   protected ready: Promise<void>;
+  protected migrations: MigrationOperation[];
 
-  constructor(db: ConnectionType, opts: Partial<MigratorOptionType> = {}) {
+  constructor(
+    db: ConnectionType,
+    opts: Partial<MigratorOptionType> = {},
+    migrations: MigrationOperation[],
+  ) {
     this.options = {...defaultSessionStorageMigratorOptions, ...opts};
     this.connection = db;
+    this.migrations = migrations;
```

Comment:
Should we validate the migrations at this point?

Reply:
How so? What sort of validations can we do ?